scio-bigquery/src/main/scala/com/spotify/scio/bigquery/types/TypeProvider.scala

```diff
@@ -220,6 +222,18 @@ private[types] object TypeProvider {
       q"override def toTableRow: ($name => ${p(c, GModel)}.TableRow) = ${p(c, SType)}.toTableRow[$name]")
   }
 
+  /** Enforce macro is not enclosed by package, therefore should be enclosed by class/object etc. */
+  private def checkMacroEnclosed(c: Context): Unit = {
+    // TODO: scala 2.11
+    //if (!c.internal.enclosingOwner.isClass) {
+    // c.abort(c.enclosingPosition,
+    //  "Invalid macro application - must be applied within class/object.")
+    // }
+    if (c.enclosingClass.isEmpty) {
+      c.abort(c.enclosingPosition,
+        "Invalid macro application - must be applied within class/object.")
```

Comment:
👍 that said will skip the name, that we get from `enclosingPosition`. So the error is gonna be (for example - you click on the error and gets you to the exact line etc):
```
Error:(5, 2) @BigQueryType declaration must be inside a class or object.
@BigQueryType.fromQuery("""|SELECT activity.value.value.value.reporting_country
```